src/controller/CHIPDeviceController.cpp

```diff
@@ -1040,15 +1041,22 @@ void DeviceCommissioner::OnDeviceAttestationInformationVerification(void * conte
     }
     else
     {
-        ChipLogProgress(Controller, "Successfully validated 'Attestation Information' command received from the device.");
-        commissioner->CommissioningStageComplete(CHIP_NO_ERROR);
+        if (deviceAttestationDelegate && deviceAttestationDelegate->ShouldWaitAfterDeviceAttestation())
```

Comment:
This should probably a parameter on the commissioning delegate instead since it avoids anyone to have to implement the new virtual method and thereby go to existing behavior with no changes, unless the client wants to change it by setting the params

Reply:
The virtual method has a default implementation that returns false, which makes this part use the existing behavior with no changes. Only delegate with new flow needs to override to return true.

Or am I misunderstanding what you mean here?

Reply:
This was resolved offline - the ShouldWaitAfterDeviceAttestation() mechanism keeps existing logic intact.